[202305] Build failure on latest 202305 in target sonic-p4rt_0.0.1_amd64.deb

<!--
     If you are reporting a new issue, make sure that we do not have any duplicates
     already open. You can ensure this by searching the issue list for this
     repository. If there is a duplicate, please close your issue and add a comment
     to the existing issue instead.

     If you suspect your issue is a bug, please edit your issue description to
     include the BUG REPORT INFORMATION shown below. If you fail to provide this
     information within 7 days, we cannot debug your issue and will close it. We
     will, however, reopen it if you later provide the information.

     For more information about reporting issues, see
     https://github.com/Azure/SONiC/wiki#report-issues

     ---------------------------------------------------
     GENERAL SUPPORT INFORMATION
     ---------------------------------------------------

     The GitHub issue tracker is for bug reports and feature requests.
     General support can be found at the following locations:

     - SONiC Support Forums - https://groups.google.com/forum/#!forum/sonicproject

     ---------------------------------------------------
     BUG REPORT INFORMATION
     ---------------------------------------------------
     Use the commands below to provide key information from your environment:
     You do NOT have to include this information if this is a FEATURE REQUEST
-->

#### Description

<!--
     Briefly describe the problem you are having in a few paragraphs.
-->

Observed on hash b226ae0699b45808364863a137d27de353cff3ee.

#### Steps to reproduce the issue:
1. git clone -b 202305 git@github.com:azure/sonic-buildimage
2. cd sonic-buildimage
3. make init
4. make configure PLATFORM=mellanox
5. make target/sonic-mellanox.bin

#### Describe the results you received:

```
[ FAIL LOG START ] [ target/debs/bullseye/sonic-p4rt_0.0.1_amd64.deb ]
Build start time: Wed Aug 9 15:31:30 UTC 2023
[ REASON ] :      target/debs/bullseye/sonic-p4rt_0.0.1_amd64.deb does not exist   NON-EXISTENT PREREQUISITES: target/debs/bullseye/libswsscommon-dev_1.0.0_amd64.deb-install
[ FLAGS  FILE    ] : [mellanox amd64 bullseye bullseye]
[ FLAGS  DEPENDS ] : [mellanox amd64 bullseye bullseye]
[ FLAGS  DIFF    ] : []
make[1]: warning: -j12 forced in submake: resetting jobserver mode.
make[1]: Entering directory '/sonic/src/sonic-p4rt'
function cleanup {
# Note: make seems to hang if Bazel is still running
cd /sonic/src/sonic-p4rt/sonic-pins && bazel  shutdown
}
trap cleanup EXIT
pushd ./sonic-pins
bazel  build --fission=yes --features=per_object_debug_info --override_repository=sonic_swss_common=/sonic/src/sonic-swss-common -c opt //p4rt_app:p4rt_deb //p4rt_app:p4rt_dbg_deb
bazel  test --fission=yes --features=per_object_debug_info --override_repository=sonic_swss_common=/sonic/src/sonic-swss-common -c opt //p4rt_app/...
cd ./bazel-bin/p4rt_app
mv sonic-p4rt_0.0.1_amd64.deb sonic-p4rt-dbgsym_0.0.1_amd64.deb /sonic/target/debs/bullseye/
popd
/sonic/src/sonic-p4rt/sonic-pins /sonic/src/sonic-p4rt
/usr/share/depot_tools/vpython3: line 48: /usr/share/depot_tools/.cipd_bin/vpython3: No such file or directory
/usr/share/depot_tools/vpython3: line 48: /usr/share/depot_tools/.cipd_bin/vpython3: No such file or directory
make[1]: *** [Makefile:34: /sonic/target/debs/bullseye/sonic-p4rt_0.0.1_amd64.deb] Error 127
make[1]: Leaving directory '/sonic/src/sonic-p4rt'
[  FAIL LOG END  ] [ target/debs/bullseye/sonic-p4rt_0.0.1_amd64.deb ]
[ 01 ] [ target/docker-config-engine-bullseye.gz ]
[ 02 ] [ target/python-wheels/bullseye/sonic_platform_common-1.0-py3-none-any.whl ]
Makefile.work:602: recipe for target 'target/sonic-mellanox.bin' failed
make[1]: *** [target/sonic-mellanox.bin] Error 2
make[1]: Leaving directory '/builds2/stepanb/202305/sonic-buildimage'
Makefile:39: recipe for target 'target/sonic-mellanox.bin' failed
make: *** [target/sonic-mellanox.bin] Error 2
```


#### Describe the results you expected:

Successful build

#### Output of `show version`:
N/A
#### Output of `show techsupport`:
N/A
#### Additional information you deem important (e.g. issue happens only occasionally):

<!--
     Also attach debug file produced by `sudo generate_dump`
-->

Observed the same problem on master when building vs target.
<pre>
[ FAIL LOG START ] [ target/debs/bullseye/sonic-p4rt_0.0.1_amd64.deb ]
Build start time: Fri Aug 11 03:40:46 UTC 2023
[ REASON ] :      target/debs/bullseye/sonic-p4rt_0.0.1_amd64.deb does not exist   NON-EXISTENT PREREQUISITES: target/debs/bullseye/libswsscommon-dev_1.0.0_amd64.deb-install
[ FLAGS  FILE    ] : []
[ FLAGS  DEPENDS ] : []
[ FLAGS  DIFF    ] : []
make[1]: warning: -j8 forced in submake: resetting jobserver mode.
make[1]: Entering directory '/sonic/src/sonic-p4rt'
function cleanup {
# Note: make seems to hang if Bazel is still running
cd /sonic/src/sonic-p4rt/sonic-pins && bazel  shutdown
}
trap cleanup EXIT
pushd ./sonic-pins
bazel  build --fission=yes --features=per_object_debug_info --override_repository=sonic_swss_common=/sonic/src/sonic-swss-common -c opt //p4rt_app:p4rt_deb //p4rt_app:p4rt_dbg_deb
bazel  test --fission=yes --features=per_object_debug_info --override_repository=sonic_swss_common=/sonic/src/sonic-swss-common -c opt //p4rt_app/...
cd ./bazel-bin/p4rt_app
mv sonic-p4rt_0.0.1_amd64.deb sonic-p4rt-dbgsym_0.0.1_amd64.deb /sonic/target/debs/bullseye/
popd
/sonic/src/sonic-p4rt/sonic-pins /sonic/src/sonic-p4rt
/usr/share/depot_tools/vpython3: line 48: /usr/share/depot_tools/.cipd_bin/vpython3: No such file or directory
/usr/share/depot_tools/vpython3: line 48: /usr/share/depot_tools/.cipd_bin/vpython3: No such file or directory
make[1]: *** [Makefile:34: /sonic/target/debs/bullseye/sonic-p4rt_0.0.1_amd64.deb] Error 127
make[1]: Leaving directory '/sonic/src/sonic-p4rt'
[  FAIL LOG END  ] [ target/debs/bullseye/sonic-p4rt_0.0.1_amd64.deb ]
make: *** [slave.mk:685: target/debs/bullseye/sonic-p4rt_0.0.1_amd64.deb] Error 1
make: *** Waiting for unfinished jobs....
</pre>

> Observed the same problem on master when building vs target.
> 
> [ FAIL LOG START ] [ target/debs/bullseye/sonic-p4rt_0.0.1_amd64.deb ]
> Build start time: Fri Aug 11 03:40:46 UTC 2023
> [ REASON ] :      target/debs/bullseye/sonic-p4rt_0.0.1_amd64.deb does not exist   NON-EXISTENT PREREQUISITES: target/debs/bullseye/libswsscommon-dev_1.0.0_amd64.deb-install
> [ FLAGS  FILE    ] : []
> [ FLAGS  DEPENDS ] : []
> [ FLAGS  DIFF    ] : []
> make[1]: warning: -j8 forced in submake: resetting jobserver mode.
> make[1]: Entering directory '/sonic/src/sonic-p4rt'
> function cleanup {
> # Note: make seems to hang if Bazel is still running
> cd /sonic/src/sonic-p4rt/sonic-pins && bazel  shutdown
> }
> trap cleanup EXIT
> pushd ./sonic-pins
> bazel  build --fission=yes --features=per_object_debug_info --override_repository=sonic_swss_common=/sonic/src/sonic-swss-common -c opt //p4rt_app:p4rt_deb //p4rt_app:p4rt_dbg_deb
> bazel  test --fission=yes --features=per_object_debug_info --override_repository=sonic_swss_common=/sonic/src/sonic-swss-common -c opt //p4rt_app/...
> cd ./bazel-bin/p4rt_app
> mv sonic-p4rt_0.0.1_amd64.deb sonic-p4rt-dbgsym_0.0.1_amd64.deb /sonic/target/debs/bullseye/
> popd
> /sonic/src/sonic-p4rt/sonic-pins /sonic/src/sonic-p4rt
> /usr/share/depot_tools/vpython3: line 48: /usr/share/depot_tools/.cipd_bin/vpython3: No such file or directory
> /usr/share/depot_tools/vpython3: line 48: /usr/share/depot_tools/.cipd_bin/vpython3: No such file or directory
> make[1]: *** [Makefile:34: /sonic/target/debs/bullseye/sonic-p4rt_0.0.1_amd64.deb] Error 127
> make[1]: Leaving directory '/sonic/src/sonic-p4rt'
> [  FAIL LOG END  ] [ target/debs/bullseye/sonic-p4rt_0.0.1_amd64.deb ]
> make: *** [slave.mk:685: target/debs/bullseye/sonic-p4rt_0.0.1_amd64.deb] Error 1
> make: *** Waiting for unfinished jobs....

Fixed by #16114

--- a/sonic-slave-bullseye/Dockerfile.j2
+++ b/sonic-slave-bullseye/Dockerfile.j2
@@ -573,10 +573,6 @@ RUN cd /usr/src/gtest && cmake . && make -C /usr/src/gtest
 RUN mkdir /var/run/sshd
 EXPOSE 22
 
-# Install depot-tools (for git-retry)
-RUN git clone https://chromium.googlesource.com/chromium/tools/depot_tools.git /usr/share/depot_tools
-ENV PATH /usr/share/depot_tools:$PATH
-
 # Install dependencies for dhcp relay test
 RUN pip3 install parameterized==0.8.1
 RUN pip3 install pyfakefs

